### PR TITLE
Add STP Actions and MoveAction

### LINKED ISSUE #24: Create structure for STP logic
We need a basic implementation of our HL interface, which can be done with our old STP (Skills, Tactics, Plays) system. This module should return the Intents robots should run given the world state.

This is a large ticket that can have its work subdivided, but is left as a single ticket since it should be easier to view all these subtasks together, as they are related.

Module we need:
* PlayExecutor: Chooses what STP play to run and call that play
* Plays: Define when they are valid/applicable, and what tactics should be assigned with what priority to the available robots.
* Tactics: Define a single-robot role within the team
* Actions: Define a simple Action the robot can do, such a shoot or move

We should investigate and decide if we still want to use coroutines. If we choose to do so it will likely be asynchronous coroutines from boost for now.

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Added STP Actiona and the MoveAction using boost coroutines.

### Testing Done

Unit tests for the MoveAction

### Resolved Issues

partially resolves #24 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/CMakeLists.txt [hunk 1]
```diff
@@ -676,6 +683,29 @@ if (CATKIN_ENABLE_TESTING)
...
+            ai/intent/move_intent.cpp
+            ai/intent/intent.cpp
+            ai/world/robot.cpp
+            util/logger/custom_g3log_sinks.h
+            util/logger/init.h
+            util/timestamp.cpp
+            )
+    target_link_libraries(move_action_test ${catkin_LIBRARIES}
+            ${G3LOG}
+            ${Boost_LIBRARIES})
```
Comment (garethellis0) [inline src/thunderbots/software/CMakeLists.txt:721]: nit; can we change to:
```
    target_link_libraries(move_action_test 
            ${catkin_LIBRARIES}
            ${G3LOG}
            ${Boost_LIBRARIES}
            )
```

File: src/thunderbots/software/ai/hl/stp/action/move_action.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    Robot robot;
+    Point destination;
+    Angle final_orientation;
+    double final_speed;
+    double close_to_dest_threshold;
+
+    // Variables to sore the Intent the Action wants to run, and the coroutine pull_type
+    std::unique_ptr<Intent> curr_intent;
+    intent_coroutine::pull_type intent_sequence;
+};
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/action/move_action.h]: end of file

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada:

File: src/thunderbots/software/CMakeLists.txt (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/CMakeLists.txt:721]: fixed

Comment (MathewMacDougall) [reply src/thunderbots/software/CMakeLists.txt:721]: fixed

File: src/thunderbots/software/ai/hl/stp/action/move_action.h (hunk 2 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/action/move_action.h]: fixed

**Review by garethellis0 — DISMISSED** [review]

**Review by garethellis0 — APPROVED** [review]

**Pull request merged**